enhancements/baremetal/baremetal-provisioning-optional.md

```diff
@@ -0,0 +1,179 @@
+---
+title: baremetal-provisioning-optional
+authors:
+  - "@stbenjam"
+reviewers:
+  - "@dhellman"
+  - "@hardys"
+  - "@kirankt"
+  - "@sadasu"
+approvers:
+  - "@hardys"
+creation-date: 2020-05-21
+last-updated: 2020-06-23
+status: provisional
+see-also:
+  - "/enhancements/baremetal/baremetal-provisioning-config.md"
+  - "/enhancements/installer/connected-assisted-installer.md"
+---
+
+# Baremetal Optional Provisioning Network
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Today the baremetal IPI platform takes a "batteries included" approach
+that has a dedicated provisioning network, HTTP caches, DHCP and TFTP
+servers, fully managed provisioning with both PXE and virtual media, and
+some level of hardware management.
+
+In some situations it may be possible to remove the requirement for a
+provisioning network entirely, in particular in the case where
+virtual media or the assisted install mechanism is used as the
+provisioning method.
+
+Previously, we added a flag to the installer that lets a user disable
+DHCP, this enhancement builds on that by making a configurable
+provisioning network profile to allow for managed, unmanaged, and
+disabled configurations.
+
+## Motivation
+
+In order to reduce the number of boolean flags in the baremetal
+platform, the various options for the provisioning network will be
+consolidated into a single enum field.
+
+### Goals
+
+The goal of this enhancement is to allow installation of a baremetal IPI
+cluster without any provisioning network dependency.
+
+### Non-Goals
+
+We do not intend to remove the static provisioning IP's at this time,
+but instead they will need to be IP's available in the external network.
+
+## Proposal
+
+A new `ProvisioningNetwork` option will be added to the baremetal
+installer platform, that features an enum of possible values:
+
+  - `managed` (default): Fully managed provisioning networking including
+     DHCP, TFTP, etc.
+
+  - `unmanaged`: Provisioning network is still present and used, but
+     user is responsible for managing DHCP. Virtual media provisioning
+     is reccomended, but PXE is still available if required.
+
+  - `disabled`: Provisioning network is fully disabled. User may only do
+     virtual media based provisioning, or bring up the cluster using
+     assisted installation. If using power management, BMC's must be
+     accessible from the machine networks. User must provide 2 IP's on
+     the external network that are used for the provisioning services.
```

Comment:
It seems like it would be possible to infer the `disabled` state if the other network details (like CIDR) aren't given. Did you consider that approach and reject it?

Reply:
Yes good point - I've updated the "Alternatives" section to describe that approach, and why I rejected it. I'm open to considering it again, if we think that's better.

Reply:
The reasoning given below makes sense.